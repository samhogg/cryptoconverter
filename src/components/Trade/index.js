import React from "react";
import PropTypes from "prop-types";
import { TradeCurrencyPair, LoadingIndicator } from "..";

import {
  TradeTitle,
  TradeButton,
  TradeSeparator,
  TradeContainer
} from "./styledComponents";
import { get } from "http";

const Trade = ({
  btcValue,
  executeTrade,
  loading,
  updateQuoteValue,
  usdValue
}) => (
  <TradeContainer onSumbit={executeTrade}>
    <TradeTitle>Trade</TradeTitle>
    <LoadingIndicator loading={loading} />
    <TradeCurrencyPair
      currency="USD"
      onChange={updateQuoteValue}
      value={usdValue}
    />
    <TradeSeparator>To</TradeSeparator>
    <TradeCurrencyPair currency="BTC" value={btcValue} />
    <TradeButton disabled={!usdValue} onClick={executeTrade} type="submit">
      Execute Trade
    </TradeButton>
  </TradeContainer>
);

Trade.propTypes = {
  btcValue: PropTypes.number,
  executeTrade: PropTypes.func.isRequired,
  loading: PropTypes.bool,
  usdValue: PropTypes.number,
  updateQuoteValue: PropTypes.func.isRequired
};

export default Trade;
