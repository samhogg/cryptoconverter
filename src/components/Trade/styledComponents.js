import styled from "styled-components";

const TradeTitle = styled.h2`
  margin: 0 0 1rem 0;
`;

const TradeContainer = styled.form`
  padding: 1rem;
  position: relative;
`;

const TradeButton = styled.button`
  width: 100%;
  background-color: ${({ theme }) => theme.colors.primary};
  color: white;
  padding: 0.5rem;
  border-radius: 5px;
  cursor: pointer;
  transition: background-color 0.1s ease-in, box-shadow 0.1s ease-in;
  box-shadow: 0 0 3px transparent;
  border: none;
  margin-top: 0.5rem;

  &:focus {
    outline: none;
    transition: box-shadow 0.1s ease-out;
    box-shadow: 0 0 10px #0062ae;
  }

  &:hover {
    background-color: #2985cc;
    transition: background-color 0.1s ease-out;
  }

  &:active {
    background-color: #1f6499;
    transition: background-color 0.1s ease-out;
  }
`;

const TradeSeparator = styled.div`
  height: 2rem;
  line-height: 2rem;
  color: #666;
  text-transform: uppercase;
  font-size: 0.8rem;
  text-align: center;
  position: relative;
`;

export { TradeTitle, TradeButton, TradeSeparator, TradeContainer };
