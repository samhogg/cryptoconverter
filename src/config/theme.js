const theme = {
  colors: {
    primary: "#0062AE",
    error: "#be0000"
  },

  // System Font Stack
  // Adapted from https://github.com/mrmrs/css-system-fonts
  fontStack: `-apple-system, BlinkMacSystemFont,
    /* MacOS and iOS */ "avenir next", avenir, /* MacOS and iOS */ "Segoe UI",
    /* Windows */ "lucida grande", /* Older MacOS */ "helvetica neue", helvetica,
    /* Older MacOS */ "Fira Sans", /* Firefox OS */ roboto, noto,
    /* Google stuff */ "Droid Sans", /* Old Google stuff */ cantarell, oxygen,
    ubuntu, /* Linux stuff */ "franklin gothic medium", "century gothic",
    /* Windows stuff */ "Liberation Sans", /* Linux */ sans-serif; /* Everything else */`
};

export default theme;
