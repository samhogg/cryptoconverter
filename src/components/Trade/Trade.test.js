import React from "react";
import renderer from "react-test-renderer";
import Trade from ".";
import { ThemeProvider } from "styled-components";
import theme from "../../config/theme";

it("Renders with reasonable defaults", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Trade
          btcValue={100}
          executeTrade={() => {}}
          loading={false}
          usdValue={1000}
          updateQuoteValue={() => {}}
        />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
