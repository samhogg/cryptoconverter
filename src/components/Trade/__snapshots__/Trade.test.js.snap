// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with reasonable defaults 1`] = `
<form
  className="styledComponents__TradeContainer-kAVRwM ZpDir"
>
  <h2
    className="styledComponents__TradeTitle-chyUqE iUFiLO"
  >
    Trade
  </h2>
  <div
    className="LoadingIndicator__LoadingContainer-eMCbod fjLIAU"
  >
    <div
      className="LoadingIndicator__LoadingDot-kbVGSy eVeAmw"
    />
    <div
      className="LoadingIndicator__LoadingDot-kbVGSy eVeAmw"
    />
    <div
      className="LoadingIndicator__LoadingDot-kbVGSy eVeAmw"
    />
  </div>
  <input
    className="TradeCurrencyPair__TradeInput-bFnQSV eEMErO"
    disabled={false}
    onChange={[Function]}
    placeholder="USD"
    readOnly={false}
    value={1000}
  />
  <div
    className="styledComponents__TradeSeparator-faTACV gxVESP"
  >
    To
  </div>
  <input
    className="TradeCurrencyPair__TradeInput-bFnQSV eEMErO"
    disabled={true}
    onChange={undefined}
    placeholder="BTC"
    readOnly={true}
    value={100}
  />
  <button
    className="styledComponents__TradeButton-ilQyaa ezRzaG"
    disabled={false}
    onClick={[Function]}
    type="submit"
  >
    Execute Trade
  </button>
</form>
`;
